src/libs/actions/Task.js

```diff
@@ -0,0 +1,139 @@
+    const assigneeChatReportID = ReportUtils.getChatByParticipants([assignee]).reportID;
+
+    // Optimistically created Action on the parent
+    const optimisticCreatedAction = ReportUtils.buildOptimisticCreatedReportAction(parentReportID);
+
+    // Create the task report
+    const optimisticTaskReport = ReportUtils.buildOptimisticTaskReport(currentUserEmail, assignee, parentReportID, optimisticCreatedAction.reportActionID, title, description);
+
+    // Create the CreatedReportAction on the task
+    const optimisticTaskCreatedAction = ReportUtils.buildOptimisticCreatedReportAction(optimisticTaskReport.reportID);
+
+    // AddCommentReportAction on the parent chat report
+    const AddCommentText = ` created a task: ${title}`;
+    Report.addComment(parentReportID, AddCommentText);
+
+    const optimisticData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.SET,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${optimisticTaskReport.reportID}`,
+            value: optimisticTaskReport,
+        },
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${optimisticTaskReport.reportID}`,
+            value: {[optimisticTaskCreatedAction.reportActionID]: optimisticTaskCreatedAction},
+        },
+    ];
+
+    const successData = [];
+
+    const failureData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${optimisticTaskReport.reportID}`,
+            value: null,
+        },
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${optimisticTaskReport.reportID}`,
+            value: {[optimisticTaskCreatedAction.reportActionID]: {pendingAction: null}},
+        },
+    ];
+
+    API.write(
+        'CreateTask',
+        {
+            // parentReportActionID: optimisticCreatedAction.reportActionID,
+            parentReportActionID: '',
+            parentReportID,
+            taskReportID: optimisticTaskReport.reportID,
+            title,
+            description,
+            assignee,
+            assigneeChatReportID,
+        },
+        {optimisticData, successData, failureData},
+    );
+
+    clearOutTaskInfo();
+
+    Navigation.navigate(ROUTES.getReportRoute(optimisticTaskReport.reportID));
+}
+
+function setDetailsValue(title, description) {
+    // This is only needed for creation of a new task and so it should only be stored locally
+    Onyx.merge(ONYXKEYS.TASK, {title, description});
+}
+
+function setTitleValue(title) {
+    Onyx.merge(ONYXKEYS.TASK, {title});
+}
+
+function setDescriptionValue(description) {
+    Onyx.merge(ONYXKEYS.TASK, {description});
+}
+
+function setAssigneeValue(assignee) {
+    let newChat = {};
+    const chat = ReportUtils.getChatByParticipants([assignee]);
+    if (!chat) {
+        newChat = ReportUtils.buildOptimisticChatReport([assignee]);
```

Comment:
This is where we should call openReport if the chat doesn't exist, should take care of everything for us

Reply:
Otherwise, the chat should already exist

Reply:
So, we only have to call `Report.openReport` if the chat does _not_ already exist? If so, that hasn't been my understanding, I thought we were opening the report in both scenarios?

Reply:
You might be right - though from my understanding openReport will create a report if it doesn't exist. Otherwise the report should already exist for the user. Though we can run through some testing today and see

Reply:
Good question - you can't share it inside a new chat that doesn't exist yet

Only for the task assignee can you share it with someone you don't already chat with (which will also create the chat report for you to share it in), but otherwise the place you share it to will have to already exist